server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java

```diff
@@ -172,7 +173,7 @@ public ReadyToBuild bucketPrefix(String bucketPrefix) {
 
             public ObjectStorageBlobConfiguration build() {
                 if (payloadCodecFactory == PayloadCodecFactory.AES256) {
-                    aesSalt.filter(s -> !s.isEmpty())
+                    aesSalt.filter(Predicate.not(Strings::isNullOrEmpty))
```

Comment:
globally, do you have any good reason to change isEmpty by isNullOrEmpty? I see several cases where it can break things, so it no more a [refactor] for me

Reply:
It covers more cases, the additional cases (NPE) should have been handled correctly previously.

Reply:
more cases are different cases, so it's no more a refactoring, it's a bug fix, and it should be tested.